src/stream/src/executor/dispatch.rs

```diff
@@ -128,8 +129,33 @@ pub fn new_output(
     addr: HostAddr,
     actor_id: ActorId,
     down_id: ActorId,
+    metrics: Arc<StreamingMetrics>,
+    actor_output_buffer_monitor_tasks: &mut HashMap<ActorId, JoinHandle<()>>,
 ) -> Result<Box<dyn Output>> {
     let tx = context.take_sender(&(actor_id, down_id))?;
+
+    let tx_clone = tx.clone();
+    // monitor backpressure rate
+    let task = tokio::spawn(async move {
+        let actor_id_str = actor_id.to_string();
+        const REPORT_FREQUENCY: usize = 15;
+        loop {
+            let mut bp_cnt = 0;
+            for _ in 0..REPORT_FREQUENCY {
+                // Buffer being fully occupied means backpressure happened.
```

Comment:
This sounds more like "how many times we observe there's a back pressure", instead of the actual back pressure time.

It is possible that data chunks come like a pulse -- within 1 second, the buffer is back pressured for some moments because downstream cache miss, and then all data pass very quickly. If the observation is always done at 1 second boundary, this metrics won't be accurate. Also note our expected checkpoint interval is 250ms for now.

I think this metrics is not accurate enough to reflect the system status. But if you think that's enough for scaling, you can use it for now.

Reply:
Does Flink collect metrics the same way as you've done?